### PR TITLE
feat: 🎨 iterate assets and resources

### LINKED ISSUE #8: Publish GitHub release with DuckDB table
Basically, copy some of the GH Actions I was using before into the new one.

### PR DESCRIPTION
**Pull request by davidgasquez**
With this PR, I'm trying to both simplify the assets code as well as move to best practices. 

These are some of the changes:

- Use a resource for the new Allo Indexer GraphQL Endpoint
- Use native Dagster retry logic 
- Redo dbt models

Closes #8, #39, #51.

### CODE REVIEW

**Comment by davidgasquez** [comment]
I might be bringing their API down :see_no_evil: 

```
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://grants-stack-indexer-v2.gitcoin.co/graphql
```

**Comment by davidgasquez** [comment]
@DistributedDoge, sorry for the large PR. I don't know what got me :sweat_smile:

Mind giving a super quick review? There are a lot of changes, hopefully trying to make things simpler.

File: ggdp/assets/allo.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                nodeId
+                projectId
+                recipientAddress
+                roundId
+                tokenAddress
+                transactionHash
+            }
+        }
+        """
+    response = indexer_graphql.paginated_query(query, size=50000)
```
Comment (DistributedDoge) [inline ggdp/assets/allo.py]: Page size seems to be a bit on larger side, but if API can handle that and so can GH runner, I have no strong objections.

File: ggdp/resources.py [hunk 2]
```diff
@@ -65,3 +65,15 @@ def upload_csv(self, df: pd.DataFrame, name: str) -> requests.Response:
         response.raise_for_status()
 
         return response
+
+
+class GrantsStackIndexerGraphQL(ConfigurableResource):
+    ENDPOINT: str = "https://grants-stack-indexer-v2.gitcoin.co/graphql"
+
+    def query(self, query: str, variables: dict = {}):
```
Comment (DistributedDoge) [inline ggdp/resources.py:78]: Is `query` method a good place to add `retry` decorator? 

At the moment if single fetch fails here, so does the parent asset using this resource and so does entire pipeline.

**Comment by DistributedDoge** [comment]
Step in good direction, and great refactor!

- support one-big-push so that allo-assets swap to indexer-V2.
- like the split into `allo` and `other` assets categories.
- we got more data, and easier query logic so props to Allo team here.

Concurrency limit (preferably by `asset-tag` so only Allo resources are affected) could help to make sure we don't hammer indexer API beyond what it can handle:

https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines

I also don't know how stable the indexer V2 API is under best circumstances, so error handling injected into `GrantsStackIndexerGraphQ` class could help?

**Comment by davidgasquez** [comment]
> We got more data, and easier query logic so props to Allo team here.

Yes! :rocket: Having the GraphQL endpoint helps a lot. 

> Concurrency limit (preferably by asset-tag so only Allo resources are affected) could help to make sure we don't hammer indexer API beyond what it can handle.

I think the issue is not as much the concurrency of all assets but the `donations` one hammering the API when paginating. Making the requests a bit smaller and perhaps adding some jitter between will help.

> I also don't know how stable the indexer V2 API is under best circumstances, so error handling injected into GrantsStackIndexerGraphQ class could help?

Definitely! Error handling + smart retrying needs to be there.

---

Got a beautiful green DAG after running `make clean && make run`.

image

File: ggdp/assets/allo.py (hunk 1 above)
Comment (davidgasquez) [reply ggdp/assets/allo.py]: Made it 10.000! :see_no_evil: 

Still on the large size.

File: ggdp/resources.py (hunk 2 above)
Comment (davidgasquez) [reply ggdp/resources.py:78]: :100:

**Pull request merged**